### ISSUE TITLE
Intermittent error when launching modals (TypeError: this.dialog.showModal is not a function)

### ISSUE THREAD

**Issue by charlesp-siemens** [siemens/ix#983]
### What happened?

Hi,
I recently opened a bug relating to the errors
`TypeError: this.dialog.showModal is not a function`
and
`TypeError: Cannot read properties of null (reading 'showModal')`
The issue I raised before was related specifically to Jest tests failing with this error.  https://github.com/siemens/ix/issues/946.

This ticket was closed because the issue could not be recreated, for whatever reason. We are **still** finding it in ours tests on our end, but after taking another look we've noticed we **also** get it during runtime of our application, and it causes some weird rendering issues with the dialog that appears.

When the error occurs, the modal is rendered on launch without a background blur, and sometimes without the ability to interact with any of the buttons. E.g:
![image](https://github.com/siemens/ix/assets/151835310/c6cf27ea-b27c-4313-bce3-31dabba71b57)

We've tried testing consistent recreation steps but all we've found is that it seems to occur more often when the size of the viewport is changed between launches of the modal?

Do properly demonstrate, I've recorded the first few seconds of me opening and closing a modal in the _completely unchanged_ Stackblitz example listed in the Ix Documentation: 
(https://ix.siemens.io/docs/controls/modal, 
https://stackblitz.com/edit/3d7byw?file=src%2Fmodal.tsx). 
While doing so I also managed to make a different but similar error appear (`Cannot read properties of null: reading 'id'`).

https://github.com/siemens/ix/assets/151835310/775d0738-ceed-4147-9181-dc02da28e840

### What type of frontend frameware are you seeing the problem on?

React

### Which version of iX do you use?

2.0.4

### Code to produce this issue.

```shell
Link to unchanged Ix Modal example Stackblitz: https://stackblitz.com/edit/3d7byw?file=src%2Fmodal.tsx.
You can of course also open the Stackblitz from the Ix Documentation Modal page, after scrolling down to the react example: https://ix.siemens.io/docs/controls/modal.
```

**Comment by jul-lam** [comment]
Thank you for opening another issue. The previous one was closed, since the issue couldn't be reproduced with the given information.

**Comment by jul-lam** [comment]
As indicated in the code provided in the response to the now-resolved issue #946 and in the [documentation](https://ix.siemens.io/docs/controls/modal#react), the use of `IxApplicationContext` is recommended. This should address your issue. I intend to create a PR that adds `IxApplicationContext` to the preview example. However, you should be able to resolve your issue without the need to wait for the new version that will include this PR.

If the issue persists, feel free to open a new issue linked to this one, since this issue will be closed by the PR fixing the preview-example.
